controllers/users.js

```diff
@@ -362,18 +362,27 @@ const getSelfDetails = async (req, res) => {
 const updateSelf = async (req, res) => {
   try {
     const { id: userId } = req.userData;
+    const { user } = await dataAccess.retrieveUsers({ id: userId });
+    // eslint-disable-next-line no-console
+    console.log("usrt", req.userdata);
+
     if (req.body.username) {
-      const { user } = await dataAccess.retrieveUsers({ id: userId });
       if (!user.incompleteUserDetails) {
         return res.boom.forbidden("Cannot update username again");
       }
       await userQuery.setIncompleteUserDetails(userId);
     }
 
-    const user = await userQuery.addOrUpdate(req.body, userId);
+    if (req.body.roles) {
+      if (user && user.roles.in_discord && !user.roles.developer) {
+        return res.boom.forbidden("Cannot update roles");
+      }
+    }
+
+    const updatedUser = await userQuery.addOrUpdate(req.body, userId);
 
-    if (!user.isNewUser) {
-      // Success criteria, user finished the sign up process.
+    if (!updatedUser.isNewUser) {
+      // Success criteria, user finished the sign-up process.
       userQuery.initializeUser(userId);
       return res.status(204).send();
```

Comment:
Aren't we sending any message in the response?

Reply:
That's not under this PR I have only changed the variable name after you look at this function I think you'll be more clear on what I have done